libraries/Microsoft.Bot.Connector/Authentication/AdalAuthenticator.cs

```diff
@@ -217,6 +221,9 @@ private bool IsAdalServiceUnavailable(Exception ex)
             return adalServiceException.ErrorCode == MsalTemporarilyUnavailable || adalServiceException.StatusCode == 429;
         }
 
+        private bool IsAdalServiceInvalidRequest(Exception ex)
+            => ex is AdalServiceException adal && adal.StatusCode == 400;
```

Comment:
do we only need to care about 400's or 4xx in general or perhaps not 2xx? is there any documentation around it that you refereed for this?

Reply:
I verified the 400 for Arlington bot requesting against Fairfax resource app.

I have not yet verified the status code for Fairfax bot requesting against the Arlington resource app.

I didn't see documentation, unfortunately.